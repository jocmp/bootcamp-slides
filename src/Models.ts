export interface SlideshowModel {
    id: number;
    title: string;
    slides: SlideModel[];
}

export interface SlideModel {
    title: string;
    slide_type: string;
    content?: string[];
}

export interface AppState {
    slideshow: SlideshowModel;
}