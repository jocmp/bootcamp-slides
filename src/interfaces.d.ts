declare module 'enzyme-to-json'