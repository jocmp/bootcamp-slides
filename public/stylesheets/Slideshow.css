.slideshow-container {
  width: 55%;
  margin-right: 20%;
  display: block;
  float: left; }

.slideshow {
  height: 60vh;
  vertical-align: center;
  margin-top: 10px;
  border: 1px solid #4C4845; }

.directional-button, .next-button, .previous-button {
  display: inline-block;
  padding: 12px 20px;
  font-size: 12pt;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  outline: none;
  color: #fff;
  background-color: #A5488B;
  border: none;
  border-radius: 12px;
  box-shadow: 0 8px #999;
  margin-top: 8px; }
  .directional-button:active, .next-button:active, .previous-button:active {
    background-color: #7F386B;
    box-shadow: 0 5px #666;
    transform: translateY(4px); }
  .directional-button:hover, .next-button:hover, .previous-button:hover {
    background-color: #7F386B; }

.next-button {
  background-color: #16CBC4;
  float: right; }
  .next-button:hover {
    background-color: #0E7F7B; }

.previous-button {
  float: left;
  background-color: white;
  border: 1px solid #CCC;
  color: #333; }
  .previous-button:hover {
    background-color: #E6E6E6; }
