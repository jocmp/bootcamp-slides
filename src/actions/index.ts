import { SlideshowProps } from '../Props'

export const nextSlide = (props: SlideshowProps) => ({
    type: 'SET_NEXT_SLIDE',
    currentIndex: nextIndex(props), 
    slideshow: props.slideshow
})

const nextIndex = (props: SlideshowProps): number => {
    const index: number = props.currentIndex;
    const length: number = props.slideshow.slides.length;
    return index < length - 1 ? index + 1 : index; 
}