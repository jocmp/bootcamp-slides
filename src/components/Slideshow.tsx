import * as React from 'react'
import Slide from './Slide'

const slideIndices: number[] = [1, 2, 3]
const Slideshow = () => {
    return (
        <ul>
        {slideIndices.map(index =>
            <Slide slideIndex={index}/>
        )}
        </ul>
    );
}

export default Slideshow;