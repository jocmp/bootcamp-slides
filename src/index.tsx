import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './components/App';
import './index.css';
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

export const slideshow = (state: any, action: any) => {
  switch(action.type) {
    case 'LOAD_SLIDESHOW':
      return {
        slideshow: action.slideshow
      }
    default:
      return {};
  }
}

export const store = createStore(slideshow, applyMiddleware(thunk));
store.subscribe(() => console.log(store.getState()))

ReactDOM.render(
  <App store={store} />,
  document.getElementById('root') as HTMLElement
);
