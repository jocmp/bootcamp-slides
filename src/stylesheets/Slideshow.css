.slideshow-container {
  width: 50%;
  margin-left: 25%;
  margin-right: 25%; }

.slideshow {
  height: 60vh;
  vertical-align: center;
  margin-top: 10px;
  border: 1px solid black; }

.next-button {
  float: right; }

.previous-button {
  float: left; }
