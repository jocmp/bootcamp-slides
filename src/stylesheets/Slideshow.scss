@import "./colors";

$button-background: #A5488B;
$button-background-highlighted: #7F386B;

.slideshow-container {
  width: 55%;
  // background: pink;
  margin-right: 20%;
  display: block;
  float: left;
}

.slideshow {
  height: 60vh;
  vertical-align: center;
  margin-top: 10px;
  border: 1px solid $black;
 }

.directional-button {
  // https://www.w3schools.com/css/tryit.asp?filename=trycss_buttons_animate3
  display: inline-block;
  padding: 12px 20px;
  font-size: 12pt;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  outline: none;
  color: #fff;
  background-color: $button-background; 
  border: none;
  border-radius: 12px;
  box-shadow: 0 8px #999;
  margin-top: 8px;

  &:active {
    background-color: $button-background-highlighted;
    box-shadow: 0 5px #666;
    transform: translateY(4px);
  }

  &:hover {
    background-color: $button-background-highlighted;
  }
}

.next-button {
  @extend .directional-button;
  background-color: $teal;
  float: right;

  &:hover {
    background-color: $teal-highlighted; 
  }
}

.previous-button {
  @extend .directional-button;
  float: left;
  background-color: white;
  border: 1px solid #CCC;
  color: #333;

  &:hover {
    background-color: #E6E6E6;
  }
}

.invisible {
  visibility: hidden;
}