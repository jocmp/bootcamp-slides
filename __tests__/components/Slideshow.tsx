import * as React from 'react'
import Slideshow from '../../src/components/Slideshow'
import { SlideshowModel, SlideModel } from '../../src/Models'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Slideshow', () => {
    const slide: SlideModel = {
        title: "Exciting Slide",
        slide_type: "title",
    }; 
    const show: SlideshowModel = {
        id: 1,
        title: "Exciting Presentation",
        slides: [ slide, slide, slide ]
    };
    
    it('renders', () => {
        const wrapper = shallow(<Slideshow slideshow={show} currentIndex={0} handleNextClick={jest.fn()} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('responds to next click', () => {
        const nextListener = jest.fn();
        const wrapper = shallow(<Slideshow slideshow={show} currentIndex={0} handleNextClick={nextListener} />)
        wrapper.find('.next-button').simulate('click') // optionally: stubbed out text edit event
        expect(nextListener).toHaveBeenCalled()
    });
});