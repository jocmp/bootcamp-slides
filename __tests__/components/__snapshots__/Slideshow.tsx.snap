exports[`Slideshow renders 1`] = `
<div>
  <TitleSlide
    slide={
      Object {
        "slide_type": "title",
        "title": "Exciting Slide",
      }
    } />
  <button
    className="next-button"
    onClick={[Function]}>
    Next
  </button>
</div>
`;
